install_cluster.py

```diff
@@ -0,0 +1,260 @@
+
+            config_path = join(REMOTE_CONFIG_DIR,
+                               '{}_config.yaml'.format(instance.name))
+            instance.run_command(
+                'cp {0} /etc/cloudify/config.yaml'.format(config_path))
+
+            logger.info('Installing %s', instance.name)
+            instance.run_command(
+                'cfy_manager install --private-ip {0} --public-ip {1}'.format(
+                    instance.private_ip, instance.public_ip))
+
+
+def _sort_instances_dict(instances_dict):
+    for _, instance_items in instances_dict.items():
+        if len(instance_items) > 1:
+            instance_items.sort(key=lambda x: int(x.name.rsplit('-', 1)[1]))
+
+
+def _get_instances_dict(config):
+    instances_dict = {instance_type: [] for instance_type in INSTANCES_TYPES}
+    username = config.get('machine_username')
+    for node_name, node_dict in config.get('existing_vms').items():
+        new_vm = CfyNode(node_dict.get('private_ip'),
+                         node_dict.get('public_ip'),
+                         JUMP_HOST_SSH_KEY_PATH,
+                         username,
+                         node_name)
+        instances_dict[new_vm.node_type].append(new_vm)
+    _sort_instances_dict(instances_dict)
+    return instances_dict
+
+
+def _create_install_cluster_directory():
+    logger.info('Creating `remote_cluster_install` directory')
+    run(['rm', '-rf', '{}_old'.format(LOCAL_INSTALL_CLUSTER_DIR)])
+    if exists(LOCAL_INSTALL_CLUSTER_DIR):
+        run(['mv', LOCAL_INSTALL_CLUSTER_DIR,
+             '{0}_old'.format(LOCAL_INSTALL_CLUSTER_DIR)])
+    os.mkdir(LOCAL_INSTALL_CLUSTER_DIR)
+    copy(JUMP_HOST_LICENSE_PATH,
+         join(LOCAL_INSTALL_CLUSTER_DIR, 'license.yaml'))
+
+
+def _show_manager_ips(manager_nodes):
+    managers_str = ''
+    for manager in manager_nodes:
+        managers_str += '{0}: {1}\n'.format(manager.name, manager.public_ip)
+    logger.info('In order to connect to one of the managers, use one of the '
+                'following IPs:\n%s', managers_str)
+
+
+def _prepare_config_files(instances_dict, load_balancer_ip):
+    os.mkdir(join(LOCAL_INSTALL_CLUSTER_DIR, 'config_files'))
+    _prepare_postgresql_config_files(instances_dict)
+    rabbitmq_credentials = (_random_credential_generator(),
+                            _random_credential_generator())
+    _prepare_rabbitmq_config_files(instances_dict, rabbitmq_credentials)
+    _prepare_manager_config_files(instances_dict, rabbitmq_credentials,
+                                  load_balancer_ip)
+
+
+def main():
+    config = get_dict_from_yaml(JUMP_HOST_CONFIG_PATH)
+    load_balancer_ip = config.get('load_balancer_ip')
+    rpm_download_link = config.get('manager_rpm_download_link')
+    instances_dict = _get_instances_dict(config)
+    _create_install_cluster_directory()
+
+    logger.info('Downloading Cloudify Manager')
+    run(['curl', '-o', JUMP_HOST_RPM_PATH, rpm_download_link])
```

Comment:
As you have other code in to support running where this has previously been run: What happens if curl -o hits a collision? Overwrite? Nice error? Download with a .1 suffix, thus potentially introducing a subtle bug here?

Reply:
Also, whether any action needs taking here will depend not just on curl's behaviour, but also on the intent behind allowing re-runs on the same system. We should discuss this.